### PR TITLE
[specs] add channel networks

### LINKED ISSUE #1409: [specs] nonceExpiry not documented
https://specs.counterfactual.com/en/latest/09-install-virtual-app-protocol.html is out-of-date and does not include the `nonceExpiry` parameter

### PR DESCRIPTION
**Pull request by IIIIllllIIIIllllIIIIllllIIIIllllIIIIll**
Gives a high-level overview of the current virtual app mechanism. Briefly explains the design goals, limitations and roadmap. fixes #1409

### CODE REVIEW

File: packages/specs/source/channel-networks.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Channel Networks
+
+The framework currently contains basic support for channel networks via "virtual apps". In this design, an app instance between that set of parties can have some state deposit assigned indirectly to it, without the set of parties having a direct channel containing them. Since the app instance is not directly funded, no commitment to `StateChannelTransaction.sol:executeAppConditionalTransaction` is made to that app instance. Instead, a number of commitments to `TwoPartyVirtualEthAsLump` are made in different direct channels to the same app instance (i.e., passing the same `appIdentityHash`). The commitment passes the following struct as a field.
+
+
+```solidity
+  struct Agreement {
```
Comment (snario) [inline packages/specs/source/channel-networks.md]: remove indents

File: packages/specs/source/channel-networks.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
+# Channel Networks
+
+The framework currently contains basic support for channel networks via "virtual apps". In this design, an app instance between that set of parties can have some state deposit assigned indirectly to it, without the set of parties having a direct channel containing them. Since the app instance is not directly funded, no commitment to `StateChannelTransaction.sol:executeAppConditionalTransaction` is made to that app instance. Instead, a number of commitments to `TwoPartyVirtualEthAsLump` are made in different direct channels to the same app instance (i.e., passing the same `appIdentityHash`). The commitment passes the following struct as a field.
```
Comment (snario) [inline packages/specs/source/channel-networks.md]: You should rephrase to define this notion. "Indirectly funded" could be _defined_ as this

Comment (snario) [inline packages/specs/source/channel-networks.md]: `sol:executeAppConditionalTransaction` is outdated.

Comment (IIIIllllIIIIllllIIIIllllIIIIllllIIIIll) [reply packages/specs/source/channel-networks.md]: Disagree - I think it is more consistent to define "indirectly funded" as "funded but not directly funded", where "funded" is a primitive/axiomatic concept, which means that this description is a specific way that an app is "indirectly funded"

Comment (snario) [reply packages/specs/source/channel-networks.md]: I just mean to say we should name this specific construction of having two apps funded via two `TwoPartyVirtualEthAsLump.delegateTarget` function calls

Comment (snario) [reply packages/specs/source/channel-networks.md]: Let's do that in your spike for this thursday.

**Pull request merged**